### PR TITLE
APIMF-2867: improve new interfaces

### PR DESCRIPTION
**Pull request by pope1838**

### CODE REVIEW

File: amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// TODO: ARM remove private[amf]
+private[amf] class AMLClient(protected override val configuration: AMLConfiguration) extends AMFClient(configuration) {
+
+  override implicit val exec: ExecutionContext = configuration.resolvers.executionContext.executionContext
+
+  override def getConfiguration: AMLConfiguration = configuration
+  def parseDialect(url: String): Future[AMLDialectResult] = AMFParser.parse(url, configuration).map {
+    case AMFResult(d: Dialect, r) => new AMLDialectResult(d, r)
+    case other =>
+      throw InvalidBaseUnitTypeException(other.bu.meta.`type`.headOption.map(_.iri()).getOrElse(""),
```
Comment (tomsfernandez) [inline amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala]: Would probably hide the creation of this exception in a private method to avoid polluting that simple method.

File: amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                                         DialectModel.`type`.head.iri())
+  }
+
+  def parseDialectInstance(url: String): Future[AMLDialectInstanceResult] = AMFParser.parse(url, configuration).map {
```
Comment (tomsfernandez) [inline amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala:21]: Same as above. Why define this exception?

File: amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/AMLInstancePlugin.scala [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package amf.plugins.document.vocabularies
+
+import amf.client.remod.amfcore.plugins.{HighPriority, PluginPriority}
+import amf.client.remod.amfcore.plugins.parse.{AMFParsePlugin, ParsingInfo}
+import amf.core.Root
```
Comment (tomsfernandez) [inline amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/AMLInstancePlugin.scala:5]: What does this have to do with the PR?

Comment (pope1838) [reply amf-aml/shared/src/main/scala/amf/plugins/document/vocabularies/AMLInstancePlugin.scala:5]: Nothing, but as the Nico's PR is not merged yet, I added to build the skeleton of some AMLConfiguration methods.

File: amf-aml/shared/src/test/scala/amf/dialects/DialectInstancesParsingTest.scala [hunk 4]
```diff
@@ -904,112 +904,75 @@ trait DialectInstancesParsingTest extends DialectTests {
...
-      "instance.yaml",
-      config.golden,
-      VocabularyYamlHint,
-      target = Amf,
-      renderOptions = Some(config.renderOptions),
-      directory = s"$basePath/node-mapping-extension-with-id-templates/"
-    )
-  }
-
-  multiGoldenTest("Array property mapping with single string element", "instance.%s") { config =>
```
Comment (AgustinBettati) [inline amf-aml/shared/src/test/scala/amf/dialects/DialectInstancesParsingTest.scala]: Some tests are being deleted here!

File: amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala (hunk 1 above)
Comment (tomsfernandez) [reply amf-aml/shared/src/main/scala/amf/client/environment/AMLClient.scala]: Solved

**Review by tomsfernandez — APPROVED** [review]
